BlueBrain/nexus-web
# 4641 // Pin history version to fix issue with opening ids with encoded characters

Fixes [#4641](https://app.zenhub.com/workspaces/blue-brain-nexus-60ace035034cae00105c9307/issues/gh/bluebrain/nexus/4641)

## Description

This is meant to be a temporary solution until we update react-router to v6 (where the following issue is fixed with some caveats).
    
 If a resource id has some encoded chars (like - https://hello.lol/https%3A%2F%2Fencoded.url%2Fwow),
    
 `history.push` pushes this to window.location:
  
 > https%3A%2F%2Fhello.lol%2Fhttps%3A%2F%2Fencoded.url%2Fwow`
    
instead of:

> https%3A%2F%2Fhello.lol%2Fhttps%253A%252F%252Fencoded.url%252Fwow

Using an older version of `history` and forcing all fusion dependencies to use this version is working for now. 
I tried pinning version `4.7` of history but that didn't work.
This solution is taken from https://github.com/remix-run/history/issues/505#issuecomment-431452320
Thanks to @bilalesi for sharing the solution

## How to test this PR

1. Run `yarn` to install the older version of `history`

2. With fusion using dev version of nexus, try going to http://localhost:8000/copies/sscx/resources/https%3A%2F%2Fdev.nise.bbp.epfl.ch%2Fnexus%2Fv1%2Fresources%2Fcopies%2Fsscx%2F_%2F1bd3d9a5-7533-4033-b575-1823862c7b8c  This should work

3. If you search for the above resource in MyData or Data-Explorer or project listing views and click on its row, then it should open the resource container with the right resource instead of showing an error. 


## Types of changes


## Checklist:
